gobblin-api/src/main/java/gobblin/configuration/SourceState.java

```diff
@@ -67,6 +69,14 @@ public SourceState(State properties, List<WorkUnitState> previousTaskStates) {
     this.previousTaskStates.addAll(previousTaskStates);
   }
 
+  public List<WatermarkInterval> getPreviousWatermarkIntervals() {
```

Comment:
At least for the kafka adapter, this method is not useful, since it only returns a list of previous watermarks from previousTaskStates, but we don't know which watermark interval corresponds to which kafka partition.

It will be better to return `Map<WorkUnitState, WatermarkInterval>`.